scheduler/scheduler.go

```diff
@@ -15,31 +15,36 @@ import (
 	"github.com/pkg/errors"
 )
 
-type TaskPlanner func(string, *distro.Distro, []task.Task, bool) ([]task.Task, error)
+type TaskPlannerOptions struct {
+	ID               string
+	IsSecondaryQueue bool
+	StartedAt        time.Time
+}
+
+type TaskPlanner func(*distro.Distro, []task.Task, TaskPlannerOptions) ([]task.Task, error)
 
-func PrioritizeTasks(id string, d *distro.Distro, tasks []task.Task, isSecondaryQueue bool) ([]task.Task, error) {
+func PrioritizeTasks(d *distro.Distro, tasks []task.Task, opts TaskPlannerOptions) ([]task.Task, error) {
```

Comment:
Should distro also be a part of the opts?

Reply:
I'd rather not.